R/waapwlspetpeese.R

```diff
@@ -0,0 +1,610 @@
+    summaryPeese <- .wwppPeese(.maGetInputEs(dataset, options), .maGetInputSe(dataset, options))
+
+    # take care of the transformed estimates
+    if (options[["measures"]] == "correlation") {
+      summaryPet   <- .wwppTransformEstimates(summaryPet,   options)
+      summaryPeese <- .wwppTransformEstimates(summaryPeese, options)
+    }
+
+    models[["object"]] <- list(
+      pet   = summaryPet,
+      peese = summaryPeese
+    )
+
+  }
+
+  return()
+}
+.wwppPet                     <- function(y, se) {
+
+  lmPet      <- lm(y ~ se, weights = 1/se^2)
+  summaryPet <- as.data.frame(summary(lmPet)$coefficients)
+  summaryPet <- .wwppAddCi(summaryPet)
+
+  row.names(summaryPet)[2]  <- "PET"
+  attr(summaryPet, "df")    <- length(y) - 2
+  attr(summaryPet, "sigma") <- sigma(lmPet)
+  attr(summaryPet, "fit")   <- lmPet
+  attr(summaryPet, "y")     <- y
+  attr(summaryPet, "se")    <- se
+
+  return(summaryPet)
+}
+.wwppPeese                   <- function(y, se) {
+
+  lmPeese      <- lm(y ~ I(se^2), weights = 1/se^2)
+  summaryPeese <- as.data.frame(summary(lmPeese)$coefficients)
+  summaryPeese <- .wwppAddCi(summaryPeese)
+
+  row.names(summaryPeese)[2]  <- "PEESE"
+  attr(summaryPeese, "df")    <- length(y) - 2
+  attr(summaryPeese, "sigma") <- sigma(lmPeese)
+  attr(summaryPeese, "fit")   <- lmPeese
+  attr(summaryPeese, "y")     <- y
+  attr(summaryPeese, "se")    <- se
+
+  return(summaryPeese)
+}
+.wwppWls                     <- function(y, se) {
+
+  lmWls      <- lm(y ~ 1, weights = 1/se^2)
+  summaryWls <- as.data.frame(summary(lmWls)$coefficients)
+  summaryWls <- .wwppAddCi(summaryWls)
+
+  attr(summaryWls, "df")    <- length(y) - 1
+  attr(summaryWls, "sigma") <- sigma(lmWls)
+  attr(summaryWls, "fit")   <- lmWls
+  attr(summaryWls, "y")     <- y
+  attr(summaryWls, "se")    <- se
+
+  return(summaryWls)
+}
+.wwppWaap                    <- function(y, se) {
+
+  summaryWls <- .wwppWls(y, se)
+  powered    <- abs(summaryWls["(Intercept)", "Estimate"] / 2.8) >= se
+
+  if(sum(powered) >= 2){
+    summaryWaap <- .wwppWls(y[powered], se[powered])
+  }else{
+    summaryWaap <- list()
+  }
+
+  attr(summaryWaap, "nPowered") <- sum(powered)
+
+  return(summaryWaap)
+}
+# create the tables
+.wwppMakeTestsTables         <- function(jaspResults, dataset, options, type) {
+
+  if (!is.null(jaspResults[["fitTests"]])) {
+    return()
+  } else {
+    # create container
+    fitTests <- createJaspContainer(title = gettext("Model Tests"))
+    fitTests$position <- 1
+    fitTests$dependOn(.wwppDependencies)
+    jaspResults[["fitTests"]] <- fitTests
+  }
+
+  models   <- jaspResults[["models"]]$object
+
+  ### test of effect
+  effectTest <- createJaspTable(title = gettext("Test of Effect"))
+  effectTest$position <- 1
+  fitTests[["effectTest"]] <- effectTest
+
+  effectTest$addColumnInfo(name = "type",  title = "",            type = "string")
+  effectTest$addColumnInfo(name = "stat",  title = gettext("t"),  type = "number")
```

Comment:
These things do not need a `gettext()` right? `t` and `p` that is

Reply:
yes, removed